articles/working-with-pythons-unittest/unittests.md

```diff
@@ -0,0 +1,221 @@
+In software engineering, a unit test refers to a program used to automatically check for bugs in individual parts/units of software. Unit testing is an important phase of the software development life cycle.
+To carry out a test, a developer specifies a set of testcases and their expected results for comparison. Most programming languages provide inbuilt unit testing frameworks. In this article, we will focus on Python's unittest.
+Regression testing refers to retesting software to ensure that it works well after a change has been made (after refactoring code). In this article, we'll see how we can use the unittest module to perform regression testing.
+
+### Prerequisites
+You will need a basic understanding of the working Python functions and classes to get started.
+
+#### Setup
+1. Open your favourite editor. I use Visual Studio Code.
+   Alternatively, you can find a working implementation of the project [here](https://replit.com/@StephenMwangi1/unit-testing) which you can execute by clicking the run button. You can also click the fork button to create a copy which you can modify.
+2. Make sure you have Python3 installed. If not, install it from [here](https://www.python.org/downloads/).
+   Note that the Python standard library provides the `unittest` module hence no installation is required.
+3. Create two files: `mathfns.py` and `main.py` using the project structure shown below.
+
+```
+project-root
+    |
+    --- mathfns.py
+    --- main.py
+```
+
+### Example
+
+#### Program
```

Comment:
There is no need for Example and Program headers. Instead, you can add something like "Let's code" (not a must).